### ISSUE TITLE
understanding estimates

### ISSUE THREAD

**Issue by brianjmurrell** [vergoh/vnstat#113]
When I run `vnstat -i eth0` I get:

```
Database updated: 2018-11-23 13:13:18

   eth0 since 18-11-05

          rx:  142.24 GiB      tx:  26.70 GiB      total:  168.94 GiB

   monthly
                     rx      |     tx      |    total    |   avg. rate
     ------------------------+-------------+-------------+---------------
       Nov '18    142.24 GiB |   26.70 GiB |  168.94 GiB |    1.45 Tbit/s
     ------------------------+-------------+-------------+---------------
     estimated    351.61 PiB |   66.00 PiB |  417.60 PiB |

   daily
                     rx      |     tx      |    total    |   avg. rate
     ------------------------+-------------+-------------+---------------
     yesterday      1.83 GiB |    1.11 GiB |    2.93 GiB |  291.49 kbit/s
         today      4.23 GiB |    1.05 GiB |    5.28 GiB |  952.96 kbit/s
     ------------------------+-------------+-------------+---------------
     estimated      7.68 GiB |    1.91 GiB |    9.59 GiB |
```

What are those estimates of?

How do I get a projection of my use for the current month (relative to `MonthRotate`) given the month's existing (at least, if not factoring in previous month's usages also to account for increase trends, etc.) current usage?

**Comment by vergoh** [comment]
The estimates should be for the currently ongoing month and day accordingly to the table. The daily estimate and average rate looks correct, at least by a quick look. However, the monthly estimate and average rate are clearly wrong. Most likely this is related to the changes `MonthRotate` value which would suggest there's a but somewhere waiting to be fixed.

What's the value for `MonthRotate` you have set?

**Comment by brianjmurrell** [comment]
I upgraded to 2.0.

**Comment by vergoh** [comment]
Ok, I've managed to replicate the issue and it doesn't appear to be anything related to the database content or the 1.x -> 2.0 upgrade. It's enough to change `MonthRotate` to any other value than 1 for `vnstat` command alone to cause the monthly estimate and the traffic rate for the current month to get multiplied by some rather high value. The value of `MonthRotate` doesn't change the error as long as the value is higher than 1. The estimate and the average rate are calculated on the fly so the content of the database itself doesn't appear to be affected.

**Comment by vergoh** [comment]
I had a Raspberry Pi running with the changes from 777d462 with `MonthRotate` set to 2 in order to validate the fix really did what is was supposed to do also in a real environment. Past midnight the month did change correctly but the estimates still didn't fully look right:
```
Database updated: 2018-12-02 02:05:00

   wlan0 since 13/02/18

          rx:  676.71 MiB      tx:  89.23 MiB      total:  765.94 MiB

   monthly
                     rx      |     tx      |    total    |   avg. rate
     ------------------------+-------------+-------------+---------------
       Nov '18    197.28 MiB |   13.12 MiB |  210.40 MiB |   680.00 bit/s
       Dec '18    599.66 KiB |  531.72 KiB |    1.10 MiB |   30.89 kbit/s
     ------------------------+-------------+-------------+---------------
     estimated      5.10 GiB |    4.52 GiB |    9.63 GiB |

   daily
                     rx      |     tx      |    total    |   avg. rate
     ------------------------+-------------+-------------+---------------
     yesterday      6.00 MiB |    5.96 MiB |   11.96 MiB |    1.16 kbit/s
         today    599.66 KiB |  531.72 KiB |    1.10 MiB |    1.24 kbit/s
     ------------------------+-------------+-------------+---------------
     estimated      6.75 MiB |    5.98 MiB |   12.73 MiB |
```
The monthly section should have shown the same average rate as the daily section during the first day of the month based on the `MonthRotate` setting. I tracked the issue to a part of older code handling timezones and the issue has now been fixed in 1baf8bb making the average rates also equal.

**Comment by lidalong50226** [comment]
[root@localhost ~]# vnstat -m

 ens160  /  monthly

        month        rx      |     tx      |    total    |   avg. rate
     ------------------------+-------------+-------------+---------------
       2025-03    265.77 GiB |  466.80 GiB |  732.56 GiB |    2.35 Mbit/s
       2025-04    550.97 GiB |  754.63 GiB |    1.28 TiB |   16.16 Mbit/s
     ------------------------+-------------+-------------+---------------
     estimated      2.01 TiB |    2.75 TiB |    4.76 TiB |
[root@localhost ~]# vnstat -v
vnStat 2.9 by Teemu Toivola <tst at iki dot fi>
The os is RockyLinux 8.10
